### PR TITLE
Permissions - Make implied permissions declarative

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This improves the implied-permission system to make it simpler for us to say that one permission implies another.

Before
----------------------------------------
Implied permissions were off in their own little space.

After
----------------------------------------
Every permission declaration can include an 'implies' key which tells the permission system which sub-permissions it includes.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@colemanw I think tests are highlighting some other magic permission you haven't taken into account

**Comment by colemanw** [comment]
@eileenmcnaughton fixed.

**Comment by eileenmcnaughton** [comment]
might be good for @totten to eyeball that last patch

File: CRM/Core/Permission.php [hunk 1]
```diff
@@ -668,6 +694,10 @@ public static function getCorePermissions() {
       'administer CiviCRM' => [
         'label' => $prefix . ts('administer CiviCRM'),
         'description' => ts('Perform all tasks in the Administer CiviCRM control panel and Import Contacts'),
+        'implies' => [
+          'administer CiviCRM system',
+          'administer CiviCRM data',
+        ],
```
Comment (totten) [inline CRM/Core/Permission.php:700]: That looks a lot nicer than how it was before. 👍

**Comment by colemanw** [comment]
@totten's eyeball didn't seem to have a problem with this PR so I guess that means it's merge ready?

**Comment by eileenmcnaughton** [comment]
Yeah - that only needed a light pass by Tim in case of anything that we hadn't thought of

**Pull request merged**